crates/ruff_linter/src/rules/pyupgrade/rules/native_literals.rs

```diff
@@ -244,7 +245,22 @@ pub(crate) fn native_literals(
 
             let arg_code = checker.locator().slice(arg);
 
-            let content = match (parent_expr, literal_type, has_unary_op) {
+            let mut needs_space = false;
+            // Look for the `Rpar` token of the call expression and check if there is a keyword token right
+            // next to it without any space separating them. Without this check, the fix for this
+            // rule would create a syntax error.
+            // Ex) `bool(True)and None` no space between `)` and the keyword `and`.
+            if let Ok(token_idx) =
+                tokens.binary_search_by(|token| token.range().end().cmp(&call_range.end()))
```

Comment:
Would it be possible to use `after` or `in_range` here over implementing the binary search manually